xdg-ninja.sh

```diff
@@ -124,53 +109,89 @@ auto_set_decoder() {
     else
         if has_command bat; then
             DECODER="bat -pp --decorations=always --color=always --language markdown"
-            loudprintf "Markdown rendering will be done by bat. (Glow is recommended)\n"
+            if [ "$QUIET" != false ]; then
+                printf "Markdown rendering will be done by bat. (Glow is recommended)\n"
+            fi
         elif has_command pygmentize; then
             DECODER="pygmentize -l markdown"
-            loudprintf "Markdown rendering will be done by pygmentize. (Glow is recommended)\n"
+            if [ "$QUIET" != false ]; then
+                printf "Markdown rendering will be done by pygmentize. (Glow is recommended)\n"
+            fi
         elif has_command highlight; then
             DECODER="highlight --out-format ansi --syntax markdown"
-            loudprintf "Markdown rendering will be done by highlight. (Glow is recommended)\n"
+            if [ "$QUIET" != false ]; then
+                printf "Markdown rendering will be done by highlight. (Glow is recommended)\n"
+            fi
         else
-            loudprintf "Markdown rendering not available. (Glow is recommended)\n"
-            loudprintf "Output will be raw markdown and might look weird.\n"
+            if [ "$QUIET" != false ]; then
+                printf "Markdown rendering not available. (Glow is recommended)\n"
+                printf "Output will be raw markdown and might look weird.\n"
+            fi
+        fi
+        if [ "$QUIET" != false ]; then
+            printf "Install glow for easier reading & copy-paste.\n"
         fi
-        loudprintf "Install glow for easier reading & copy-paste.\n"
     fi
 }
 
-auto_set_decoder
-set_colors
+set_colors() {
+    case "$COLOR" in
+        "always")
+            return
+            ;;
+        "auto")
+            if [ -t 1 ] && [ "$NO_COLOR" != false ]; then # Check if used in a pipe or if the NO_COLOR env variable is set.
+                return
+            fi
+            ;;
+    esac
+    DECODER="cat"
```

Comment:
As I said, cat shouldn't be used as default decoder, since external processes shouldn't be spawned without a need